compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -756,6 +847,7 @@ export class Environment {
     return (
       moduleName.toLowerCase() === 'react' ||
       moduleName.toLowerCase() === 'react-dom' ||
+      [...this.#globals.keys()].includes(moduleName) ||
```

Comment:
why do we need this?

Reply:
This predicate gates giving types to imports from globals [here](https://github.com/facebook/react/pull/30670/files/6be688d88d66e16701d5fab08c364d40e4dcbdbb#diff-3ebfb2fef1f4c7802f493c87477cce82d5a1385b5c39a887db981a4308e2e23dR813-R816). Gonna rename this function to `isKnownTypedModule`!